### PR TITLE
Create plans for the future development.

### PR DESCRIPTION
**Pull request by wendavid552**
Add about what we should learn in the future, mainly about the algorithms and data structres.
Any problems about it? Anything to be added into it?

### CODE REVIEW

File: Plans/201109.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  - Basic
+  - 强连通分量，点双连通分量，边双连通分量
+  - 2-SAT
+  - 
+- Matching
+  - Its algorithms are not so important as we can use the algorithm of network flows as a substitution
+- Network Flows
+  - Basic Algorithm（Dinic，ISAP）
+  - 基本模型
+  -
```
Comment (IcecreamArtist) [inline Plans/201109.md]: shortest path

File: Plans/201109.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- Network Flows
+  - Basic Algorithm（Dinic，ISAP）
+  - 基本模型
+  - 
+## Number Theory
+- 整除性，不定方程，同余式，同余方程组
+- Miller-Rabin Test与Pollard-rho等素数相关算法
+- Euler$\Phi$函数，Mobius反演
+- 平方剩余，勒让德记号，原根
+## Dynamic Programming
```
Comment (IcecreamArtist) [inline Plans/201109.md:60]: 概率dp、树形dp等（dp应该范围很广）

File: Plans/201109.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- 凸包及其性质
+- 杂题
+  
+## Other programming(e.g. Greedy, Linear Programming, some optimized version of direct ways like branch cutting)
+  - 贪心
+  - 单纯形
+  - $\alpha-\beta$剪枝
+  
+## Some Techniques
+- Middle-attack / Meet in the middle
```
Comment (IcecreamArtist) [inline Plans/201109.md:78]: 模拟退火、莫队等

Comment (wendavid552) [reply Plans/201109.md:78]: 模拟退火总感觉很少用到吧，莫队倒是可以加入上面的数据结构特殊算法里面

File: Plans/201109.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+- BFS & DFS
+  - Basic
+  - 强连通分量，点双连通分量，边双连通分量
+  - 2-SAT
+  - **How to write DFS in artificial stacks (Not that provided by system)**
+- Shortest Path & MST
+- Matching
+  - Its algorithms are not so important as we can use the algorithm of network flows as a substitution
+- Network Flows
+  - Basic Algorithm（Dinic，ISAP）
```
Comment (Fashionfx) [inline Plans/201109.md:52]: 最小费用最大流

File: Plans/201109.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+## Data Structres(Offline, or some special ways to modify)
+- 树链剖分
+- 动态树LCT
+- cdq分治
+- 整体二分
+- 分块
+- 莫队算法（分块变体）
+- 可持久化
+
```
Comment (Fashionfx) [inline Plans/201109.md:22]: rmq算法

**Pull request merged**